File: rest/model/nftTransfer.js
// SPDX-License-Identifier: Apache-2.0

class NftTransfer {
  /**
   * Parses nft_transfer from element in transaction.nft_transfer jsonb column
   */
  constructor(nftTransfer) {
    this.isApproval = nftTransfer.is_approval;
    this.receiverAccountId = nftTransfer.receiver_account_id;
    this.senderAccountId = nftTransfer.sender_account_id;
    this.serialNumber = nftTransfer.serial_number;
    this.tokenId = nftTransfer.token_id;
  }

  static IS_APPROVAL = `is_approval`;
  static RECEIVER_ACCOUNT_ID = `receiver_account_id`;
  static SENDER_ACCOUNT_ID = `sender_account_id`;
  static SERIAL_NUMBER = `serial_number`;
  static TOKEN_ID = `token_id`;
}

export default NftTransfer;

File: common/src/main/java/org/hiero/mirror/common/domain/token/NftTransfer.java
// SPDX-License-Identifier: Apache-2.0

package org.hiero.mirror.common.domain.token;

import jakarta.persistence.Convert;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hiero.mirror.common.converter.EntityIdConverter;
import org.hiero.mirror.common.domain.entity.EntityId;

@AllArgsConstructor(access = AccessLevel.PRIVATE) // for Builder
@Builder
@Data
@NoArgsConstructor
public class NftTransfer {

    public static final long WILDCARD_SERIAL_NUMBER = -1;

    private Boolean isApproval;

    @Convert(converter = EntityIdConverter.class)
    private EntityId receiverAccountId;

    @Convert(converter = EntityIdConverter.class)
    private EntityId senderAccountId;

    private Long serialNumber;

    @Convert(converter = EntityIdConverter.class)
    private EntityId tokenId;
}
